port/linux/omrvmem.c

```diff
@@ -1601,7 +1669,7 @@ getMemoryInRangeForDefaultPages(struct OMRPortLibrary *portLibrary, struct J9Por
 		Trc_PRT_vmem_omrvmem_reserve_memory_ex_UnableToAllocateWithinSpecifiedRange(byteAmount, startAddress, endAddress);
 
 		memoryPointer = NULL;
-	} else {
+	} else if (0 == (mode & OMRPORT_VMEM_MEMORY_MODE_MMAP_HUGE_PAGES)) {
```

Comment:
Looking to `madvise()` manual: the only limitation I can find is "Transparent Huge Pages work only with private anonymous pages". How this check corresponds with such statement?

Reply:
I would imagine this call to be redundant if heap will be allocated using huge pages, because we would be reserving huge pages in the first place right? Here we would check if huge pages are being used, if it is, then we wouldn't need to call `madvise` with `MADV_HUGEPAGE`

Reply:
Yes, I understand this. And check is correct. However I am trying to understand (may be it is academic) would mapped huge pages be "private anonymous"? Would it be incorrect to call `madvise()` in this case or just unnecessary?

Reply:
Oh I see, I would imagine that `madvise` wouldn't do anything. I just checked and `madvise` still returns true/success even though huge pages is being used to reserve memory. That was true for when heap was "shared and not anonymous" (double mapping enabled), and for when it was "private and anonymous" (double mapping disabled). Not sure if that helps with your question